python/tk_multi_workfiles/file_save_form.py

```diff
@@ -181,7 +186,21 @@ def _on_use_next_available_version_toggled(self, checked):
 
     def _start_preview_update(self):
         """
+        Starts the path preview task if we're not initializing the gui. If a preview task is
+        currently running, it will be stopped and a new one will be launched.
         """
+
+        # When initializing the gui, events are fired multiple times due to signals
+        # emitted from the widgets. Here we are muting the start_preview_update call
+        # to avoid creating and deleting the preview task multiple times, which is not only
+        # wasteful but also makes the code harder to debug since 5 tasks end up computing
+        # the preview.
+        if self._initializing:
+            return
+
+        # Disable the button while the path is computed.
+        self._disable_save("Please wait while Toolkit calculates the next available file name.")
```

Comment:
I'm not sure this is needed as generate_path is called again before the save operation (https://github.com/shotgunsoftware/tk-multi-workfiles2/pull/3/files#diff-1d0c9979f5f3e3247a4b3cb32ee3445aR664) so even if the user hits save straight away before the preview has been generated everything should be ok.  The main problem with greying out the button is that it tends to flicker as users click around and feels a bit rubbish!

It's up to you guys but personally, as long as the preview generation is fixed (the include_dirty=not clean_only above) then I'd leave this out.

The only other thing to make it even more robust would be to perform a complete search (rather than using the cached result) when the user hits save.  

There is also an additional check in the save logic to see if the file exists before it's actually overwritten - currently that the user can choose to overwrite the file but it could be changed to just return to this dialog and force them to try again...